docs/zkapps/tutorials/10-account-updates.mdx

```diff
@@ -0,0 +1,378 @@
+
+### Smart Contracts
+
+You will build two smart contracts to use in your example:
+
+First, [ProofsOnlyZkApp.ts](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/10-account-updates/src/ProofsOnlyZkApp.ts).
+
+You will configure this zkApp to be modifiable only by using proofs. For this example, the zkApp is not upgradable after it is deployed. This means that while someone must own the private key to deploy the zkApp itself - after it is deployed, only the rules in the zkApp determine how the zkApp functions. The private key is no longer useful for anything.
+
+This zkApp also has methods that call other methods, so we can explore how that impacts a transaction's AccountUpdates.
+
+Start by adding the main contents of the zkApp:
+
+```typescript
+
+export class ProofsOnlyZkApp extends SmartContract {
+  @state(Field) num = State<Field>();
+  @state(Field) calls = State<Field>();
+
+  deploy(args: DeployArgs) {
+    super.deploy(args);
+    this.setPermissions({
+      ...Permissions.default(),
+      setDelegate: Permissions.proof(),
+      setPermissions: Permissions.proof(),
+      setVerificationKey: Permissions.proof(),
+      setZkappUri: Permissions.proof(),
+      setTokenSymbol: Permissions.proof(),
+      incrementNonce: Permissions.proof(),
+      setVotingFor: Permissions.proof(),
+      setTiming: Permissions.proof(),
+    });
+  }
+
+  @method init() {
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertFalse();
+
+    super.init();
+    this.num.set(Field(1));
+    this.calls.set(Field(0));
+  }
+
+  ...
+```
+
+This code configures the zkApp as described and initializes the zkApp with the values you want. 
+
+:::caution
+Note you assert that `provedState` is `false` in `init()` to ensure that `init()` cannot be called again after the zkApp is set up. Without this, your zkApp could be reset by anyone calling the `init()` method on your zkApp and is recommended for most zkApps. 
+:::
+
+Next, add two functions:
+
+```typescript
+  ...
+
+  @method add(incrementBy: Field) {
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertTrue();
+
+    const num = this.num.get();
+    this.num.assertEquals(num);
+    this.num.set(num.add(incrementBy));
+
+    this.incrementCalls();
+  }
+
+  @method incrementCalls() {
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertTrue();
+
+    const calls = this.calls.get();
+    this.calls.assertEquals(calls);
+    this.calls.set(calls.add(Field(1)));
+  }
+
+  ...
+```
+
+:::caution
+In each of these methods, you are also asserting `provedState` is `true` to ensure the zkApp was initialized as expected because `provedState` becomes true after `init` is called and is recommended for most zkApps.
```

Comment:
```suggestion
In each of these methods, you are also asserting `provedState` is `true` to ensure the zkApp was initialized as expected because `provedState` becomes true after `init()` is invoked, and is recommended for most zkApps.
```